realm/realm-library/src/androidTest/java/io/realm/internal/CollectionTests.java

```diff
@@ -398,13 +399,16 @@ public void onChange(Collection collection1) {
                     case 1:
                         assertEquals(collection1.size(), 5);
                         sharedRealm.close();
-                        looperThread.testComplete();
+                        break;
+                    default:
+                        fail();
```

Comment:
If it reached here after `testComplete` called, the test will still fail with assertion.